Lib/test/test_codecs.py

```diff
@@ -3415,5 +3415,19 @@ def test_rot13_func(self):
             'To be, or not to be, that is the question')
 
 
+class CodecNameNormalizationTest(unittest.TestCase):
+    """Test the normalizestring function via codecs module"""
+    def test_normalized_encoding(self):
+        def search_function(encoding):
+            if encoding == "aaa_8":
+                return (1, 2, 3, 4)
+            else:
+                return (None, None, None, None)
+        codecs.register(search_function)
+        self.assertEqual((1, 2, 3, 4), codecs.lookup('AAA-8'))
```

Comment:
It would be nice to test more variants than just "AAA-8". What about whitespace ("AAA 8"), dot ("AAA.8"), non-ASCII character, etc.? What about multiple separators like "AAA-----8" or "AAA    8"?

Reply:
make sense. I will update it.